optimize/upgrade/src/main/java/io/camunda/optimize/upgrade/plan/factories/UpgradePlanFactory.java

```diff
@@ -7,9 +7,46 @@
  */
 package io.camunda.optimize.upgrade.plan.factories;
 
+import io.camunda.optimize.service.db.DatabaseClient;
+import io.camunda.optimize.service.db.schema.OptimizeIndexNameService;
+import io.camunda.optimize.service.exceptions.OptimizeRuntimeException;
+import io.camunda.optimize.upgrade.exception.UpgradeRuntimeException;
 import io.camunda.optimize.upgrade.plan.UpgradeExecutionDependencies;
 import io.camunda.optimize.upgrade.plan.UpgradePlan;
+import java.io.IOException;
 
 public interface UpgradePlanFactory {
   UpgradePlan createUpgradePlan(final UpgradeExecutionDependencies upgradeExecutionDependencies);
+
+  void logErrorMessage(final String message);
+
+  default boolean isC7LicenseDataPresent(
+      final DatabaseClient databaseClient, final OptimizeIndexNameService indexNameService) {
+    try {
+      return databaseClient.countWithoutPrefix(
+              indexNameService.getOptimizeIndexAliasForIndex("license"))
+          > 0;
+    } catch (OptimizeRuntimeException e) {
+      // This gets thrown if the index is not found, so we can safely assume that there is no data
```

Comment:
this also gets thrown on OpenSearch if _any_ exception is thrown, so I don't think this is a safe assumption that we can count on here

Reply:
Hm this was just picked over from the recent OS upgradeplan PR changes in C7, I will check with Giuliano.

Reply:
@grlimacan I added some changes to this check with [this commit](refactor: implement PR comments)[refactor: implement PR comments](https://github.com/camunda/camunda/pull/22573/commits/9ad0ae12b637641dffab576f81519846c39d5425), tried them locally on the C7 repo with OS and ES as well. Can you have a look and verify this makes sense to you, then I will pick this over to the equivalent C7 repo files too.

Reply:
@PHWaechtler Unfortunately the link to the commit above got smashed, but I looked into the changes in general and they make sense, please to pick this over to C7. Am I seeing correctly that the main difference is that you are doing the exists check additionally?